File: benchmarks/custom/client-ios/src/main/java/com/github/larchaon/ios/GraphicsViewController.java
package com.github.larchaon.ios;

import com.github.larchaon.shared.TrialSuite;
import org.robovm.apple.coregraphics.*;
import org.robovm.apple.uikit.*;
import org.robovm.objc.annotation.IBAction;
import org.robovm.objc.annotation.IBOutlet;

public class GraphicsViewController extends UIViewController implements UIControl.OnTouchUpInsideListener {

    UILabel benchmarkResult;
    UIImageView imageView;

    @Override
    public void viewDidLoad() {
        super.viewDidLoad();

        final UIButton runButton = UIButton.create(UIButtonType.RoundedRect);
        runButton.setFrame(new CGRect(0, 100, 200.0f, 37.0f));
        runButton.setTitle("Run", UIControlState.Normal);
        runButton.addOnTouchUpInsideListener(this);

        benchmarkResult = new UILabel(new CGRect(0, 150, 2300, 44));

        imageView = new UIImageView(new CGRect(0, 200, 100, 100));

        getView().addSubview(runButton);
        getView().addSubview(benchmarkResult);
        getView().addSubview(imageView);
    }

    @Override
    public void onTouchUpInside(UIControl control, UIEvent event) {
        TrialSuite suite = new TrialSuite(200);
        for (int times = 0; times < 200; times++) {

            CGRect bounds = new CGRect(CGPoint.Zero(), new CGSize(100, 100));
            float scale = 0;
            UIGraphics.beginImageContext(new CGSize(200, 200), false, scale);
            CGContext context = UIGraphics.getCurrentContext();

            context.setFillColor(CGColor.create(CGColorSpace.createDeviceRGB(), new float[]{times % 256, 256 - times % 256, 0, 1}));

            long now = System.nanoTime();

            long sum = testableMethod(context, bounds);

            long end = System.nanoTime();

            suite.addTrial(now, end);
        }
        benchmarkResult.setText(suite.getAverage() + "");
        System.out.println(suite.getComaSeparatedData());
    }

    public int testableMethod(CGContext context, CGRect bounds) {
        double w = bounds.getWidth();
        double h = bounds.getHeight();
        for(float x = 0; x < w; x+=2) {
            for(float y = 0; y < h; y+=2) {
                context.fillRect(new CGRect(x, y, 1, 1));
            }
        }

        UIImage image = UIGraphics.getImageFromCurrentImageContext();
        UIGraphics.endImageContext();
        imageView.setImage(image);
        return 0;
    }
}

File: benchmarks/ios/iOS Benchmarks/GraphicsViewController.swift
import UIKit

class GraphicsViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBOutlet weak var runBtn: UIButton!
    @IBOutlet weak var benchmarkResult: UILabel!
    @IBOutlet weak var imageView: UIImageView!
    
    @IBAction func run(sender: AnyObject) {
        self.runBtn.enabled = false;
        var array = [Float]();
            
        for (var times:Int = 0; times < 200; times++) {
            let bounds = CGRect(origin: CGPoint.zeroPoint, size: CGSize(width: 100, height: 100))
            let opaque = false
            let scale: CGFloat = 0
            UIGraphicsBeginImageContextWithOptions(CGSize(width: 200, height: 200), opaque, scale)
            let context = UIGraphicsGetCurrentContext()
                
            CGContextSetFillColorWithColor(context, CGColorCreate(CGColorSpaceCreateDeviceRGB(), [CGFloat(times % 256), CGFloat(256 - times % 256), CGFloat(0), CGFloat(1)]));
                
            let now = CACurrentMediaTime();
                
            let sum = self.testableMethod(context, bounds: bounds);
                
            let end = CACurrentMediaTime();
                
            array.append(Float(end - now)*100);
         }
         var str = "";
         for obj in array {
            str += obj.description + ", ";
         }
         NSLog("array: " + str);
         self.benchmarkResult.text = self.getAverage(array).description;
         self.runBtn.enabled = true;
    }
    
    func testableMethod(context: CGContext, bounds: CGRect) -> Int {
        var w = bounds.width;
        var h = bounds.height;
        for var x: CGFloat = 0; x < w; x+=2 {
            for var y: CGFloat = 0; y < h; y+=2 {
                CGContextFillRect(context, CGRectMake(x,y,1,1));
            }
        }
        
        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        imageView.image = image;
        return 0;
    }

    
    func getAverage(trials: [Float]) -> Float {
        var sum: Float = 0.0;
        for trial in trials {
            sum = sum + trial;
        }
        return sum / Float(trials.count);
    }
}

